zellij-server/src/lib.rs

```diff
@@ -193,23 +226,66 @@ pub fn start_server(os_input: Box<dyn ServerOsApi>, socket_path: PathBuf) {
                     .send_to_pty(PtyInstruction::NewTab)
                     .unwrap();
             }
+            ServerInstruction::AttachClient(attrs, _) => {
+                *session_state.write().unwrap() = SessionState::Attached;
+                let rlock = session_data.read().unwrap();
+                let session_data = rlock.as_ref().unwrap();
+                session_data
+                    .senders
+                    .send_to_screen(ScreenInstruction::TerminalResize(attrs.position_and_size))
+                    .unwrap();
+                let mode_info =
+                    get_mode_info(InputMode::Normal, attrs.palette, session_data.capabilities);
+                session_data
+                    .senders
+                    .send_to_screen(ScreenInstruction::ChangeMode(mode_info.clone()))
+                    .unwrap();
+                session_data
+                    .senders
+                    .send_to_plugin(PluginInstruction::Update(
+                        None,
+                        Event::ModeUpdate(mode_info),
+                    ))
+                    .unwrap();
+            }
             ServerInstruction::UnblockInputThread => {
-                os_input.send_to_client(ServerToClientMsg::UnblockInputThread);
+                if *session_state.read().unwrap() == SessionState::Attached {
+                    os_input.send_to_client(ServerToClientMsg::UnblockInputThread);
+                }
             }
             ServerInstruction::ClientExit => {
-                *sessions.write().unwrap() = None;
-                os_input.send_to_client(ServerToClientMsg::Exit);
+                *session_data.write().unwrap() = None;
+                os_input.send_to_client(ServerToClientMsg::Exit(ExitReason::Normal));
                 break;
             }
+            ServerInstruction::DetachSession => {
+                *session_state.write().unwrap() = SessionState::Detached;
+                os_input.send_to_client(ServerToClientMsg::Exit(ExitReason::Normal));
+                os_input.remove_client_sender();
+            }
             ServerInstruction::Render(output) => {
-                os_input.send_to_client(ServerToClientMsg::Render(output))
+                if *session_state.read().unwrap() == SessionState::Attached {
+                    if let Some(op) = output {
```

Comment:
Screen/Tab sends `Render(Some<String>)` to the server for forwarding it to the client. Earlier the client used to handle it and exit if it was `None` (it's `None` when the user closes the last Tab or Pane). Now, the server handles it and forwards the output if it is `Some(_)` or sends an explicit `Exit(_)` message in case of `None`.